packages/grafana-data/src/transformations/transformers/joinDataFrames.ts

```diff
@@ -362,6 +362,74 @@ function joinOuterTabular(
   return data;
 }
 
+/**
+ * This function performs a sql-style inner join on tabular data;
+ * it will combine records from two tables whenever there are matching
+ * values in a field common to both tables.
+ *
+ * NOTE: This function implicitly assumes that the first array in each AlignedData
+ * contains the values to join on. It doesn't explicitly specify a column field to join on,
+ * but rather uses the 0th position of the arrays (AlignedData[0]) to determine the joining keys.
+ * Then, when processing the tables, the function iterates over the values in the `xValues`
+ * (the joining keys) array and checks if the current row `currentRow` already includes the value.
+ * If a matching value is found, it joins the corresponding values from the remaining arrays `yValues`
+ * (all other non-joining key arrays) to create a new row in the joined table.
+ *
+ * @param {AlignedData[]} tables - The tables to join.
+ *
+ * @returns {Array<Array<string | number | null | undefined>>} The joined tables as an array of arrays, where each array represents a row in the joined table.
+ */
+function joinInner(tables: AlignedData[]): Array<Array<string | number | null | undefined>> {
```

Comment:
```suggestion
function joinInnerTabular(tables: AlignedData[]): Array<Array<string | number | null | undefined>> {
```
Should we keep naming convention the same as the outer join for tabular data? 🤔

Reply:
update: it seems that for inner we can do both series + tabular data while for some reason outer is differentiated and split out 🤔

Reply:
Correct. I was originally going to add an `INNER (TABULAR)` option, but the previous `INNER` wasn't actually behaving like an `INNER` anyway, so instead of adding yet another option, @leeoniya suggested we just correct the original `INNER`'s behavior, which I think was the right move. The reason we split the 2 `OUTER` joins - I believe - is because they behave significantly differently then the other, depending on the data (time series or SQL-like). The `INNER`, however, due to the nature of an `INNER` join itself, will behave very similarly for both data types - sans a few edge cases.